io.github.picocrypt.Picocrypt.yaml

```diff
@@ -0,0 +1,40 @@
+app-id: io.github.picocrypt.Picocrypt
+runtime: org.freedesktop.Platform
+runtime-version: "23.08"
+sdk: org.freedesktop.Sdk
+command: picocrypt-gui
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --filesystem=home
+modules:
+  - name: picocrypt-gui
+    buildsystem: simple
+    build-commands:
+      - touch io.github.picocrypt.Picocrypt.desktop
+      - echo "[Desktop Entry]" >> io.github.picocrypt.Picocrypt.desktop
+      - echo "Name=Picocrypt" >> io.github.picocrypt.Picocrypt.desktop
+      - echo "Comment=Secure file encryption tool" >> io.github.picocrypt.Picocrypt.desktop
+      - echo "Type=Application" >> io.github.picocrypt.Picocrypt.desktop
+      - echo "Exec=picocrypt-gui" >> io.github.picocrypt.Picocrypt.desktop
+      - echo "Icon=io.github.picocrypt.Picocrypt" >> io.github.picocrypt.Picocrypt.desktop
+      - echo "Categories=Utility;Security;" >> io.github.picocrypt.Picocrypt.desktop
+      - install -Dm755 Picocrypt /app/bin/picocrypt-gui
+      - install -Dm644 key.svg /app/share/icons/hicolor/scalable/apps/io.github.picocrypt.Picocrypt.svg
+      - install -Dm644 io.github.picocrypt.Picocrypt.desktop /app/share/applications/io.github.picocrypt.Picocrypt.desktop
+      - install -Dm644 io.github.picocrypt.Picocrypt.metainfo.xml /app/share/metainfo/io.github.picocrypt.Picocrypt.metainfo.xml
+    sources:
+      - type: file
+        url: https://github.com/Picocrypt/Picocrypt/releases/download/1.42/Picocrypt
```

Comment:
you should build it from source.

Reply:
> you should build it from source.

This is not very convenient, we will have to put the Golang inside for building. But the binary is already self-sufficient in itself, what prevents it from being used 'as is'?

Reply:
why is it not convenient? It's better.

Reply:
> This is not very convenient, we will have to put the Golang inside for building.

we have a go extension to take care of the toolchain https://github.com/flathub/org.freedesktop.Sdk.Extension.golang

See how other go apps do it https://github.com/flathub/dev.skynomads.Seabird/blob/master/dev.skynomads.Seabird.yaml

Reply:
> > This is not very convenient, we will have to put the Golang inside for building.
> 
> we have a go extension to take care of the toolchain https://github.com/flathub/org.freedesktop.Sdk.Extension.golang
> 
> See how other go apps do it https://github.com/flathub/dev.skynomads.Seabird/blob/master/dev.skynomads.Seabird.yaml

But how do we handle dependencies if there is no network in flatpak environment during building? 
`require (
	github.com/Picocrypt/dialog v0.0.0-20240831001746-9ca708a9cd29
	github.com/Picocrypt/giu v0.0.0-20240831005244-5771b35043ac
	github.com/Picocrypt/imgui-go v0.0.0-20240831004007-6f60d7beadf6
	github.com/Picocrypt/infectious v0.0.0-20240830233326-3a050f65f9ec
	github.com/Picocrypt/serpent v0.0.0-20240830233833-9ad6ab254fd7
	github.com/Picocrypt/zxcvbn-go v0.0.0-20240831000415-fccb38ccb913
	golang.org/x/crypto v0.26.0
)

require (
	github.com/Picocrypt/gl v0.0.0-20240831002619-6531d2bba5fc // indirect
	github.com/Picocrypt/glfw/v3.3/glfw v0.0.0-20240831003212-7f16c5fb374b // indirect
	github.com/Picocrypt/mainthread v0.0.0-20240831004314-496f638392b3 // indirect
	github.com/Picocrypt/w32 v0.0.0-20240831001500-1183079d4d57 // indirect
	golang.org/x/sys v0.24.0 // indirect
)
`

Reply:
Use the go-vendor generator script to get a dependency manifest of your go modules then add that manifest to the PR

https://github.com/flatpak/flatpak-builder-tools/blob/master/go-get/flatpak-go-vendor-generator.py

Reply:
> Use the go-vendor generator script to get a dependency manifest of your go modules then add that manifest to the PR
> 
> https://github.com/flatpak/flatpak-builder-tools/blob/master/go-get/flatpak-go-vendor-generator.py

Thanks, helped a lot